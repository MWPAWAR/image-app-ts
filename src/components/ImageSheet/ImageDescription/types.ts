import { ImageTypes } from '../../types'

export interface ImageDescriptionPropsType {
  selectedImage: ImageTypes
}
