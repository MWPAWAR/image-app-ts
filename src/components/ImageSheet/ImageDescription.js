import imageDescStyles from './ImageDescription.module.css';

const ImageDescription = ({ selectedImage }) => (
  selectedImage.description ? (
    <div className={imageDescStyles.imgDescription}>
      <div className={imageDescStyles.label}>Description</div>
      <div className={imageDescStyles.description}>{selectedImage.description}</div>
    </div>
  ) : null
);

export default ImageDescription;
