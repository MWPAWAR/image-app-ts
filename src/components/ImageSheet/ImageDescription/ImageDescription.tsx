import imageDescStyles from './ImageDescription.module.css'
import { ImageDescriptionPropsType } from './types'

const ImageDescription = ({ selectedImage }: ImageDescriptionPropsType) =>
  selectedImage.description ? (
    <div className={imageDescStyles.imgDescription}>
      <div className={imageDescStyles.label}>Description</div>
      <div className={imageDescStyles.description}>
        {selectedImage.description}
      </div>
    </div>
  ) : null

export default ImageDescription
