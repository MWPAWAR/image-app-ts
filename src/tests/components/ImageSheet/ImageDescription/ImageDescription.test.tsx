import renderer from 'react-test-renderer'
import ImageDescription from '../../../../components/ImageSheet/ImageDescription/ImageDescription'
import { imageMock1, imageMock2 } from '../../../mocks/image'

describe('ImageDescription', () => {
  it('renders ImageDescription correctly', () => {
    const tree = renderer.create(
      <ImageDescription selectedImage={imageMock1} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders null if imageDescription is empty', () => {
    const tree = renderer.create(
      <ImageDescription selectedImage={imageMock2} />
    )
    expect(tree).toMatchSnapshot()
  })
})
