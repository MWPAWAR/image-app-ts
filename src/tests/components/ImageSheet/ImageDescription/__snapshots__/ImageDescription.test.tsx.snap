// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDescription renders ImageDescription correctly 1`] = `
<div
  className="imgDescription"
>
  <div
    className="label"
  >
    Description
  </div>
  <div
    className="description"
  >
    Laboriosam eligendi inventore officia nemo. Quisquam explicabo voluptatem. Illo laborum facilis.
  </div>
</div>
`;

exports[`ImageDescription renders null if imageDescription is empty 1`] = `null`;
